src/lib/expansion/accordion-item.ts

```diff
@@ -0,0 +1,56 @@
+import {Output, EventEmitter, Input, OnDestroy, Optional, Host} from '@angular/core';
+import {UniqueSelectionDispatcher} from '../core';
+import {CdkAccordion} from './accordion';
+
+/** Used to generate unique ID for each expansion panel. */
+let nextId = 0;
+
+/**
+ * An abstract class to be extended and decorated as a component.  Sets up all
+ * events and attributes needed to be managed by a CdkAccordion parent.
+ */
+export abstract class AccordionItem implements OnDestroy {
+  /** Event emitted every time the MdAccordianChild is closed. */
+  @Output() closed = new EventEmitter<void>();
+  /** Event emitted every time the MdAccordianChild is opened. */
+  @Output() opened = new EventEmitter<void>();
+  /** Event emitted when the MdAccordianChild is destroyed. */
+  @Output() destroyed = new EventEmitter<void>();
+  /** The unique MdAccordianChild id. */
+  readonly id = `cdk-accordion-child-${nextId++}`;
+  /** Whether the MdAccordianChild is expanded. */
+  @Input() get expanded(): boolean { return this._expanded; }
+  set expanded(expanded: boolean) {
+    // Only emit events and update the internal value if the value changes.
+    if (this._expanded !== expanded) {
+      this._expanded = expanded;
+      if (expanded) {
+        this.opened.emit();
+        /**
+         * In the unique selection dispatcher, the id parameter is the id of the CdkAccordonItem,
+         * the name value is the id of the accordion.
+         */
+        let accordionId = this.accordion ? this.accordion.id : this.id;
+        this._uniqueSelectionDispatcher.notify(this.id, accordionId);
+      } else {
+        this.closed.emit();
+      }
+    }
+  }
+  private _expanded: boolean;
+
+  constructor(@Optional() @Host() public accordion: CdkAccordion,
```

Comment:
My thought was if you had a `CdkAccordion` inside of another `CdkAccordion` that you wanted to be sure you got the correct `CdkAccordion`, so we only look at the host component.  However now that you point it out I see how flawed that logic is as it will get the first one it finds on its way up the tree.